### PR TITLE
Add FFVA INT example with integrated Cyberon engine

### PR DESCRIPTION
**Pull request by lucianomartin**
Part of https://xmosjira.atlassian.net/browse/AP-423

The task below is not valid, as the code in those folders are specific to the low power FFD device.

~~In a separate PR I will try to extend to he low_power_ffd example the use of the files moved into modules/asr:~~
 - [intent_engine]( https://github.com/lucianomartin/sln_voice/tree/feature/ffva_cyberon/modules/asr/intent_engine)
 - [intent_handler](https://github.com/lucianomartin/sln_voice/tree/feature/ffva_cyberon/modules/asr/intent_handler)

### CODE REVIEW

File: examples/ffva/bsp_config/XK_VOICE_L71/platform/platform_start.c [hunk 1]
```diff
@@ -39,9 +39,7 @@ static void gpio_start(void)
 static void flash_start(void)
 {
 #if ON_TILE(FLASH_TILE_NO)
-    uint32_t flash_core_map = ~((1 << appconfUSB_INTERRUPT_CORE) | (1 << appconfUSB_SOF_INTERRUPT_CORE));
     rtos_qspi_flash_start(qspi_flash_ctx, appconfQSPI_FLASH_TASK_PRIORITY);
-    rtos_qspi_flash_op_core_affinity_set(qspi_flash_ctx, flash_core_map);
 #endif
```
Comment (shuchitak) [inline examples/ffva/bsp_config/XK_VOICE_L71/platform/platform_start.c:43]: why are these lines deleted?

File: examples/ffva/ffva_int_cyberon.cmake [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        #[[ Target ]]                   ${TARGET_NAME}
+        #[[ Boot Partition Size ]]      ${BOOT_PARTITION_SIZE}
+        #[[ Data Partition Contents ]]  ${DATA_PARTITION_FILE}
+        #[[ Dependencies ]]             ${DATA_PARTITION_FILE}
+    )
+
+    unset(DATA_PARTITION_FILE_LIST)
+    unset(DATA_PARTITION_DEPENDS_LIST)
+
+    #add_custom_target(
```
Comment (shuchitak) [inline examples/ffva/ffva_int_cyberon.cmake:192]: Do we not need the fatfs related stuff? If not then could you delete these lines?

File: examples/ffva/src/app_conf.h [hunk 3]
```diff
@@ -15,10 +15,15 @@
...
 
+#ifndef appconfINTENT_ENGINE_READY_SYNC_PORT
+#define appconfINTENT_ENGINE_READY_SYNC_PORT      18
+#endif /* appconfINTENT_ENGINE_READY_SYNC_PORT */
+
 /* Application tile specifiers */
 #include "platform/driver_instances.h"
+#define ASR_TILE_NO             FLASH_TILE_NO
 #define FS_TILE_NO              FLASH_TILE_NO
-#define AUDIO_PIPELINE_TILE_NO  MICARRAY_TILE_NO
```
Comment (shuchitak) [inline examples/ffva/src/app_conf.h:21]: Aren't mic array tile no. and flash tile no. different? Have we changed the tile on which the pipeline runs? Also, is this needed for the UA build as well?

File: examples/ffva/src/app_conf.h [hunk 4]
```diff
@@ -28,6 +33,61 @@
 /* If in channel sample format, appconfAUDIO_PIPELINE_FRAME_ADVANCE == MIC_ARRAY_CONFIG_SAMPLES_PER_FRAME*/
 #define appconfAUDIO_PIPELINE_FRAME_ADVANCE     MIC_ARRAY_CONFIG_SAMPLES_PER_FRAME
 
+/* Enable audio response output */
+#ifndef appconfAUDIO_PLAYBACK_ENABLED
+#define appconfAUDIO_PLAYBACK_ENABLED           1
```
Comment (shuchitak) [inline examples/ffva/src/app_conf.h:38]: Isn't the fatfs stuff needed for the playback?

File: tools/ci/main_examples.txt [hunk 5]
```diff
@@ -1,8 +1,9 @@
...
-ffva_int_fixed_delay    example_ffva_int_fixed_delay    Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
-low_lower_ffd_sensory   example_low_power_ffd_sensory   Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
-ffd_sensory             example_ffd_sensory             Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
-ffd_cyberon             example_ffd_cyberon             Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
-mic_aggregator_TDM      example_mic_aggregator_tdm      No   XCORE-AI-EXPLORER   xmos_cmake_toolchain/xs3a.cmake
-mic_aggregator_USB      example_mic_aggregator_usb      No   XCORE-AI-EXPLORER   xmos_cmake_toolchain/xs3a.cmake
-asrc                    example_asrc_demo               No   XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
+ffva_ua_adec_altarch            example_ffva_ua_adec_altarch            Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
+ffva_int_fixed_delay            example_ffva_int_fixed_delay            Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
+ffva_int_cyberon_fixed_delay    example_ffva_int_cyberon_fixed_delay    Yes  XK_VOICE_L71        xmos_cmake_toolchain/xs3a.cmake
```
Comment (shuchitak) [inline tools/ci/main_examples.txt:3]: Good you added the build to ci!

**Review by shuchitak — APPROVED** [review]
Few questions, rest all looks good!

File: examples/ffva/bsp_config/XK_VOICE_L71/platform/platform_start.c (hunk 1 above)
Comment (lucianomartin) [reply examples/ffva/bsp_config/XK_VOICE_L71/platform/platform_start.c:43]: Good spot!

File: examples/ffva/src/app_conf.h (hunk 3 above)
Comment (lucianomartin) [reply examples/ffva/src/app_conf.h:21]: AUDIO_PIPELINE_TILE_NO is only used in the ASR code, and it is actually the tile with the pipeline output.
@shuchitak, should I rename it to AUDIO_PIPELINE_OUTPUT_TILE_NO?
In the FFD examples the whole pipeline runs on one tile, in FFVA AEC is on tile 1 and the rest in tile 0.

Comment (shuchitak) [reply examples/ffva/src/app_conf.h:21]: hey @lucianomartin, yeah, AUDIO_PIPELINE_OUTPUT_TILE_NO would be a better name

File: examples/ffva/src/app_conf.h (hunk 4 above)
Comment (lucianomartin) [reply examples/ffva/src/app_conf.h:38]: The FATFS data is already included in the data partition with the WW model.
